static/js/settings_users.js

```diff
@@ -445,7 +448,17 @@ function confirm_deactivation(row, user_id, status_field) {
             },
         };
         const url = "/json/users/" + encodeURIComponent(user_id);
-        settings_ui.do_settings_change(channel.del, url, {}, status_field, opts);
+
+        let data = {};
+        if ($(".send_email").is(":checked")) {
+            data = {
+                email_text: $(".email_field_textarea").val(),
+                realm_name: page_params.realm_name,
+                username: user.full_name,
```

Comment:
I don't think you should need the `realm_name` or `username` parameters in this HTTP request; the server already has these details for the target user.